zerver/models.py

```diff
@@ -1165,6 +1165,16 @@ def can_subscribe_other_users(self) -> bool:
         assert self.realm.invite_to_stream_policy == Realm.POLICY_FULL_MEMBERS_ONLY
         return not self.is_new_member
 
+    def has_permission(self, policy_name: str) -> bool:
+
+        if policy_name == 'create_stream_policy':
+            return self.can_create_streams()
+
+        if policy_name == 'invite_to_stream_policy':
+            return self.can_subscribe_other_users()
+
+        raise AssertionError("Invalid policy")
```

Comment:
Sorry, this isn't what I had in mind -- what I meant is to create a common `has_permission` function, and make `self.can_create_streams()` call that function with `create_stream_policy` as the argument,